web/src/components/onboarding/OnboardingTaxId.tsx

```diff
@@ -1,25 +1,142 @@
-import { OnboardingNumericField } from "@/components/onboarding/OnboardingNumericField";
-import { ProfileFieldErrorMap, ProfileFields } from "@/lib/types/types";
-import { ReactElement } from "react";
+/* eslint-disable @typescript-eslint/no-explicit-any */
 
-interface Props {
-  onValidation: (field: ProfileFields, invalid: boolean) => void;
+import { Content } from "@/components/Content";
+import { GenericTextField } from "@/components/GenericTextField";
+import { OnboardingField, OnboardingProps } from "@/components/onboarding/OnboardingField";
+import { ProfileDataContext } from "@/contexts/profileDataContext";
+import { useConfig } from "@/lib/data-hooks/useConfig";
+import { ProfileFieldErrorMap } from "@/lib/types/types";
+import { getTaxIdFormat, setHeaderRole } from "@/lib/utils/helpers";
+import { ReactElement, useContext, useEffect, useRef, useState } from "react";
+
+interface Props extends Omit<OnboardingProps, "fieldName" | "handleChange"> {
   fieldStates: ProfileFieldErrorMap;
-  headerAriaLevel?: number;
+  splitField?: boolean;
   handleChangeOverride?: (value: string) => void;
 }
-
-export const OnboardingTaxId = (props: Props): ReactElement => {
+export const OnboardingTaxId = ({
```

Comment:
we have an outstanding refactoring ticket to rename all of the `OnboardingFieldName` fields to `ProfileFieldName` instead, so it'd be good to integrate that new pattern with brand-new fields at least